File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        n = len(nums1)
        m = len(nums2)

        j = n-m-1
        k = m-1

        for i in range(n-1,-1,-1):
            if j>=0 and k>=0:
                if nums2[k]>nums1[j]:
                    nums1[i] = nums2[k]
                    k-=1
                else:
                    nums1[i] = nums1[j]
                    j-=1
            else:
                if j>=0:
                    nums1[i] = nums1[j]
                    j-=1
                elif k>=0:
                    nums1[i] = nums2[k]
                    k-=1

File: 88-merge-sorted-array/merge-sorted-array.ts
/**
 Do not return anything, modify nums1 in-place instead.
 */
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
    let pointer1 = m-1;
    let pointer2 = n-1;
    let i = m+n-1;
    
    while (pointer2 >= 0) {
        if (pointer1 >= 0 && nums1[pointer1] > nums2[pointer2]) {
            nums1[i] = nums1[pointer1];
            pointer1--;
        } else {
            nums1[i] = nums2[pointer2];
            pointer2--;
        }
        i--;
    }
};